### DISCUSSION TITLE
No events when deployed in AWS - how to debug?

### DISCUSSION THREAD

**Discussion by davidr912** [getsentry/sentry-dotnet#891] [Q&A]
Please mark the type framework used:

- [X] Other: Plain console application

Please mark the type of the runtime used:

- [X] .NET Core

* Version: 2.0

Please mark the NuGet packages used:

- [X] Sentry

* Version: 1.2.0

My sentry is able to send events and exceptions when run locally - but when deployed I get nothing.

`sentry.io` resolves in DNS in AWS and is contactable so there is no clear reason why there would be a difference. This is running in Linux in a docker container (alpine with glibc)

There are no visible errors etc. or trace output and `IsEnabled` returns true

Is there any way to fetch errors/problems from inside of `SentrySdk` so I can diagnose what the cause is here?

**Comment by bruno-garcia** [comment]
@davidr912 Is this AWS Lambda? 
Events are sent in the background (via a worker thread). Make sure you call `SentrySdk.Close` or close the `IDisposable` that `Init` returns before the process exists.

In AWS Lambda the process might be killed or frozen before the event gets flushed out to the network. That's why the ASP.NET Core integration calls `await FlushAsync` for all requests once they complete, if you opt-in:

https://github.com/getsentry/sentry-dotnet/blob/598ab4fbdee2df3966c9bb00ae9d16b78e6bace9/src/Sentry.AspNetCore/SentryMiddleware.cs#L93-L100

**Comment by davidr912** [comment]
hey @bruno-garcia - this is actually AWS Fargate, so pretty much just running arbitrary containers for as long as you want (AFAIK no extra behaviours re: freezing or killing)

Is there still something analogous to `FlushAsync` I should be doing at a regular basis though?

**Comment by bruno-garcia** [comment]
> Make sure you call SentrySdk.Close or close the IDisposable that Init returns before the process exists.

Are you making sure the SDK is closed when the process quits?

Do you get events at all? Even if you have the process running for 5 minutes? You container has access to the Internet?
